File: app/src/main/java/com/huanchengfly/tieba/post/models/database/Block.java
package com.huanchengfly.tieba.post.models.database;

import org.litepal.crud.LitePalSupport;

public class Block extends LitePalSupport {
    public static final int CATEGORY_BLACK_LIST = 10;
    public static final int CATEGORY_WHITE_LIST = 11;
    public static final int TYPE_KEYWORD = 0;
    public static final int TYPE_USER = 1;

    private int id;
    private int category;
    private int type;
    private String username;
    private String uid;
    private String keywords;

    public String getUid() {
        return uid;
    }

    public Block setUid(String uid) {
        this.uid = uid;
        return this;
    }

    public int getCategory() {
        return category;
    }

    public Block setCategory(int category) {
        this.category = category;
        return this;
    }

    public int getId() {
        return id;
    }

    public int getType() {
        return type;
    }

    public Block setType(int type) {
        this.type = type;
        return this;
    }

    public String getUsername() {
        return username;
    }

    public Block setUsername(String username) {
        this.username = username;
        return this;
    }

    public String getKeywords() {
        return keywords;
    }

    public Block setKeywords(String keywords) {
        this.keywords = keywords;
        return this;
    }
}

File: app/src/main/java/com/huanchengfly/tieba/post/models/database/Block.kt
package com.huanchengfly.tieba.post.models.database

import com.huanchengfly.tieba.post.fromJson
import org.litepal.crud.LitePalSupport

data class Block @JvmOverloads constructor(
    val category: Int = 0,
    val type: Int = 0,
    val keywords: String? = null,
    val username: String? = null,
    val uid: String? = null,
) : LitePalSupport() {
    val id: Long = 0L
    companion object {
        const val CATEGORY_BLACK_LIST = 10
        const val CATEGORY_WHITE_LIST = 11

        const val TYPE_KEYWORD = 0
        const val TYPE_USER = 1

        fun Block.getKeywords(): List<String> {
            return keywords?.fromJson() ?: emptyList()
        }
    }
}